common/http_test.cpp

```diff
@@ -1681,3 +1681,127 @@ TEST(HttpsTest, CorrectDefaultCertificateStoreVerification) {
 	EXPECT_TRUE(client_hit_header);
 	EXPECT_TRUE(client_hit_body);
 }
+
+TEST(HttpTest, ExponentialBackoff) {
+	http::ExponentialBackoff::ExpectedInterval exp_interval;
+
+	auto duration_fmt = [](chrono::milliseconds ms) { return to_string(ms.count()) + "ms"; };
+
+	// Test with one minute maximum interval.
+	{
+		http::ExponentialBackoff backoff(chrono::minutes(1));
+		exp_interval = backoff.NextInterval();
+		ASSERT_TRUE(exp_interval) << exp_interval.error().String();
+		EXPECT_EQ(exp_interval.value(), chrono::minutes(1)) << duration_fmt(exp_interval.value());
+
+		exp_interval = backoff.NextInterval();
+		ASSERT_TRUE(exp_interval) << exp_interval.error().String();
+		EXPECT_EQ(exp_interval.value(), chrono::minutes(1)) << duration_fmt(exp_interval.value());
+
+		exp_interval = backoff.NextInterval();
+		ASSERT_TRUE(exp_interval) << exp_interval.error().String();
+		EXPECT_EQ(exp_interval.value(), chrono::minutes(1)) << duration_fmt(exp_interval.value());
+
+		exp_interval = backoff.NextInterval();
+		ASSERT_FALSE(exp_interval);
+		EXPECT_EQ(exp_interval.error().code, http::MakeError(http::MaxRetryError, "").code);
```

Comment:
No, this is a subtlety with what `code` actually is. It is an `error_condition` which consists of a code and an error category. You can't compare the error code without also comparing the category, since there is overlap in the error code values, so for example `2` could mean many different things. So `http::MaxRetryError` cannot by itself be compared with anything without wrapping it in inside the right kind of error.

Strictly speaking we could have constructed an `error_condition` directly here, and used that instead, but because of our convenience `MakeError` functions, this is easier to write.

So I guess `code` is a somewhat misleading name, but we've become used to it.